src/Geopilot.Frontend/src/components/header/header.tsx

```diff
@@ -0,0 +1,225 @@
+import { useTranslation } from "react-i18next";
+import { FC, useState } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import {
+  AppBar,
+  Avatar,
+  Box,
+  Button,
+  Divider,
+  Drawer,
+  IconButton,
+  List,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  Toolbar,
+  Typography,
+} from "@mui/material";
+import OpenInNewIcon from "@mui/icons-material/OpenInNew";
+import LoginIcon from "@mui/icons-material/Login";
+import LogoutIcon from "@mui/icons-material/Logout";
+import { useAppSettings } from "../appSettings/appSettingsInterface";
+import { useGeopilotAuth } from "../../auth";
+import { LanguagePopup } from "./languagePopup";
+import MenuIcon from "@mui/icons-material/Menu";
+
+interface HeaderProps {
+  openSubMenu: () => void;
+}
+
+const Header: FC<HeaderProps> = ({ openSubMenu }) => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const location = useLocation();
+  const { clientSettings } = useAppSettings();
+  const { user, enabled, isAdmin, login, logout } = useGeopilotAuth();
+
+  const [userMenuOpen, setUserMenuOpen] = useState<boolean>(false);
+
+  const toggleUserMenu = (newOpen: boolean) => () => {
+    setUserMenuOpen(newOpen);
+  };
+
+  const isActive = (path: string) => {
+    if (path === "") {
+      return location.pathname === "/";
+    }
+    return location.pathname.split("/").includes(path);
+  };
+  const hasSubMenu = location.pathname.startsWith("/admin");
+
+  return (
+    <>
+      <AppBar sx={{ height: "60px" }}>
+        <Toolbar
+          sx={{
+            display: "flex",
+            flexDirection: "row",
+            justifyContent: "space-between",
+          }}>
+          <Box sx={{ display: "flex", flexDirection: "row", alignItems: "center", padding: "5px 0" }}>
```

Comment:
PP: Styling in Komponenten auslagern, welche man wiederverwenden kann. z.B. HeaderMenu etc.